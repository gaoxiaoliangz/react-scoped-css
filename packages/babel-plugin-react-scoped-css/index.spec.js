// @ts-check
const babel = require("@babel/core");
const path = require("path");
const fs = require("fs");
const plugin = require("./");

const readFixtures = () => {
  const dir = path.resolve(__dirname, "./fixtures");
  const files = fs.readdirSync(dir);

  return files.map((f) => {
    return {
      filename: f,
      path: path.resolve(dir, f),
    };
  });
};

describe("babel-plugin-react-scoped-css", () => {
  readFixtures().forEach((fixture) => {
    it(fixture.filename, () => {
      const codeContent = fs.readFileSync(fixture.path, {
        encoding: "utf8",
      });
      const { code } = babel.transform(codeContent, {
        plugins: [
          [
            plugin,
            {
              hashSeed: "userSeed",
            },
          ],
        ],
        filename: fixture.filename,
      });
      expect(code).toMatchSnapshot();
    });
  });
});
