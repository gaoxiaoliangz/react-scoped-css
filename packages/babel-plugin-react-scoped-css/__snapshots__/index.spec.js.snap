// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-react-scoped-css frag.jsx 1`] = `
"import React from 'react';
import "./frag.scoped.css?scopeId=f2245a65";

const Frag = () => {
  return <div data-v-f2245a65={""}>
      <React.Fragment>
        <div className="test-frag" data-v-f2245a65={""}></div>
      </React.Fragment>
      <>
        <div data-v-f2245a65={""}>text</div>
      </>
    </div>;
};

export default Frag;"
`;

exports[`babel-plugin-react-scoped-css from-example.jsx 1`] = `
"import React from 'react';
import styled from 'styled-components';
import styles from "./Content.scoped.scss?scopeId=81d995ea";
import Grid from './Grid';
import GridForwardProps from './GridForwardProps';
const Text = styled.div\`
  color: #666;
\`;

const FromExample = props => {
  return <div className="content" data-v-81d995ea={""}>
      <h3 data-v-81d995ea={""}>Styling html tags</h3>
      <p data-v-81d995ea={""}>p tag with style</p>

      <React.Fragment>
        <div data-v-81d995ea={""}>
          <p data-v-81d995ea={""}>content wrapped with React Fragment should be fine</p>
        </div>
      </React.Fragment>

      <h3 data-v-81d995ea={""}>Using classes</h3>
      <div className="grid" data-v-81d995ea={""} />
      <div className="grid" data-v-81d995ea={""} />

      <h3 data-v-81d995ea={""}>Styling child components with css modules</h3>
      <Grid className={styles.grid} data-v-81d995ea={""} />
      <Grid className={styles.grid} data-v-81d995ea={""} />

      <h3 data-v-81d995ea={""}>Styling child components which forward data-v attributes to its root element</h3>
      <GridForwardProps className="content-grid" data-v-81d995ea={""} />
      <GridForwardProps className="content-grid" data-v-81d995ea={""} />

      <h3 data-v-81d995ea={""}>Styling with styled-components</h3>
      <Text className="text" data-v-81d995ea={""}>Some content in styled-components</Text>
    </div>;
};

export default Content;"
`;
